src/netlink.c

```diff
@@ -0,0 +1,387 @@
+#include "netlink.h"
+#include "timeout.h"
+#include "lua.h"
+#include "lauxlib.h"
+#include "compat.h"
+#include "auxiliar.h"
+#include "socket.h"
+/*=========================================================================*\
+* Internal function prototypes
+\*=========================================================================*/
+static int global_create(lua_State *L);
+static int meth_bind(lua_State *L);
+static int meth_sendto(lua_State *L);
+static int meth_send(lua_State *L);
+static int meth_receivefrom(lua_State *L);
+static int meth_receive(lua_State *L);
+static int meth_close(lua_State *L);
+static int meth_settimeout(lua_State *L);
+static int meth_gettimeout(lua_State *L);
+static int meth_getfd(lua_State *L);
+static int meth_setfd(lua_State *L);
+static int meth_getpeername(lua_State *L);
+static int meth_setpeername(lua_State *L);
+static int meth_getsockpid(lua_State *L);
+
+static const char *netlink_trybind(p_netlink nl,int grp);
+
+/* netlink object methods */
+static luaL_Reg netlink_methods[] = {
+    {"__gc",        meth_close},
+    {"__tostring",  auxiliar_tostring},
+    {"bind",        meth_bind},
+    {"close",       meth_close},
+    {"getfd",       meth_getfd},
+    {"send",        meth_send},
+    {"sendto",      meth_sendto},
+    {"receivefrom", meth_receivefrom},
+    {"receive",     meth_receive},
+    {"setfd",       meth_setfd},
+    {"settimeout",  meth_settimeout},
+    {"gettimeout",  meth_gettimeout},
+    {"setpeername", meth_setpeername},
+    {"getpeername", meth_getpeername},
+    {"getsockpid",  meth_getsockpid},
+    {NULL,          NULL}
+};
+
+/* functions in library namespace */
+static luaL_Reg func[] = {
+    {"netlink", global_create},
+    {NULL, NULL}
+};
+
+/*-------------------------------------------------------------------------*\
+* Initializes module
+\*-------------------------------------------------------------------------*/
+int netlink_open(lua_State *L) {
+    	/* create classes */
+    	auxiliar_newclass(L, "netlink{connected}", netlink_methods);
+    	auxiliar_newclass(L, "netlink{unconnected}", netlink_methods);
+    	/* create class groups */
+    	auxiliar_add2group(L, "netlink{connected}", "netlink{any}");
+    	auxiliar_add2group(L, "netlink{unconnected}", "netlink{any}");
+	
+	auxiliar_add2group(L, "netlink{connected}", "select{able}");  
+	auxiliar_add2group(L, "netlink{unconnected}", "select{able}");  
+	luaL_setfuncs(L, func, 0);
+	return 0;
+}
+
+/*-------------------------------------------------------------------------*\
+* Send data through connected netlink socket
+\*-------------------------------------------------------------------------*/
+static int meth_send(lua_State *L) {
+	p_netlink nl = (p_netlink) auxiliar_checkclass(L, "netlink{connected}", 1);
+	size_t payload_size, sent = 0;
+	const char *payload = luaL_checklstring(L, 2, &payload_size);
+	p_timeout tm = &nl->tm; 
+
+	int flags = luaL_optinteger(L, 3, 0);
+
+	struct nlmsghdr *nlh;
```

Comment:
if you are declaring and initializing, like the lines above, why do differently with `nlh`? be consistent in style.